python/lsst/ip/diffim/metrics.py

```diff
@@ -0,0 +1,158 @@
+# This file is part of ip_diffim.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+__all__ = [
+    "NumberSciSourcesMetricTask", "NumberSciSourcesMetricConfig",
+    "FractionDiaSourcesToSciSourcesMetricTask", "FractionDiaSourcesToSciSourcesMetricConfig",
+]
+
+
+import astropy.units as u
+
+from lsst.pipe.base import Struct, InputDatasetField
+from lsst.verify import Measurement
+from lsst.verify.gen2tasks import MetricTask, register
+
+
+class NumberSciSourcesMetricConfig(MetricTask.ConfigClass):
+    sources = InputDatasetField(
+        doc="The catalog of science sources.",
+        name="src",
+        storageClass="SourceCatalog",
+        dimensions={"Instrument", "Exposure", "Detector"},
+    )
+
+
+@register("numSciSources")
+class NumberSciSourcesMetricTask(MetricTask):
+    """Task that computes the number of cataloged science sources.
```

Comment:
I'm suddenly worried we shouldn't call these "science" sources--see the ticket in JIRA.